src/util/simplify_expr2.cpp

```diff
@@ -415,6 +416,195 @@ static type2tc common_arith_op2_type(expr2tc &e, expr2tc &f)
 
 expr2tc add2t::do_simplify() const
 {
+  auto Side_1 = side_1;
+  auto Side_2 = side_2;
+
+  // X + X --> X << 1
+  if(is_symbol2t(side_1) && is_symbol2t(side_2))
+    if(side_1 == side_2)
+      return shl2tc(type, side_1, from_integer(1, type));
+
+  // (A + 1) + ~B --> A - B
+  if(is_bitnot2t(Side_1) && is_add2t(Side_2))
+    std::swap(Side_1, Side_2);
+
+  if(is_add2t(Side_1) && is_bitnot2t(Side_2))
+  {
+    if(to_constant_int2t(to_add2t(Side_1).side_2).value == 1)
```

Comment:
An `is_constant_int2t` is missing here; I believe this will solve the `bad_cast` regressions.

Reply:
Tksssssss u saved me some hours of debuging 😁😁😁